test/clippy/clippy_failure_test.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```

Comment:
As an FYI, this test is not integrated into the build system; I've just been using it manually to test. See the discussion thread in the BUILD file for my attempts to have a non-manual solution.

Reply:
Cf. https://github.com/bazelbuild/bazel-integration-testing if you want to integrate it.

Reply:
Are there any examples of integration testing beyond https://github.com/bazelbuild/bazel-integration-testing/blob/master/bazel_integration_test/test_base_test.py ? I see how to set up a python integration test and run bazel commands inside, but I'm struggling to make it reference a pre-existing workspace (e.g., one that references all the `rust_clippy` rules and sources in this directory).

Reply:
Actually, I found https://github.com/hchauvin/bazel-cloud-infra/tree/587eb7c11d528148c0243eaa392ef070e44f32b8/rbs/test ... I'll try to base the clippy test off of this example.

Reply:
I've really tried integrating this, but I ran into multiple issues:

1) Following [the only non-trivial example of the integration test I've found](https://github.com/hchauvin/bazel-cloud-infra/tree/587eb7c11d528148c0243eaa392ef070e44f32b8/rbs/test)... If I create a directory `rules_rust/test/clippy/workspace`, under which I place all these files, I can glob together the sources + WORKSPACE + BUILD file as runfiles, and read them within the test. **However**, although I want to pull in the top-level repo as:

(within `rules_rust/test/clippy/workspace/WORKSPACE`)
```
local_repository(
   name = "io_bazel_rules_rust",
   path = "path_to_io_bazel_workspace_root",
)
```
This doesn't work: The execution root of this "clippy sub-workspace" cannot reference the path to the "rules rust workspace root".

I noticed that the `examples` directory of this repo also uses `local_repository`, but since it isn't loaded via runfiles, it can use relative paths. The same doesn't appear to apply here.

2) After hacking around that issue by injecting an absolute path to the rules_rust WORKSPACE (yuck), the integration test repo failed because it is using extremely old versions of Bazel, which don't seem compatible with `rules_rust`. I tried supplying the `versions` argument to `bazel_py_integration_test`, but this argument doesn't work; it only works for the pre-defined versions of Bazel (0.21.0, 0.22.0, 0.23.2), or else the rule will depend on python libraries which don't exist.

Reply:
I have an example of these issues here in a draft PR (not building):
  https://github.com/bazelbuild/rules_rust/pull/340

Reply:
I went to dive a bit more and the project bazel-integration-testing seems dead, I would suggest to not spend too much time on that.